lib/IRGen/GenDistributed.cpp

```diff
@@ -836,15 +838,35 @@ ArgumentDecoderInfo DistributedAccessor::findArgumentDecoder(
                        instance);
 
     decoder = instance.claimNext();
-  }
 
-  auto *decodeSIL = IGM.getSILModule().lookUpFunction(SILDeclRef(decodeFn));
-  auto *fnPtr = IGM.getAddrOfSILFunction(decodeSIL, NotForDefinition,
-                                         /*isDynamicallyReplaceable=*/false);
+    /// When using library evolution functions have another "dispatch thunk"
+    /// so we must use this instead of the decodeFn directly.
+   if (classDecl->hasResilientMetadata()) {
+      if (getMethodDispatch(decodeFn) == swift::MethodDispatch::Class) {
+        SILDeclRef(decodeFn).getOverriddenVTableEntry().getDecl()->getDeclContext()->dumpContext();
```

Comment:
I suspect you meant to remove this call to `dumpContext()`?

Reply:
Yes sorry removed now! And simplified the branches a bit -- thank you both @hborla and @xedin for review 🙏